extensions/amp-consent/0.1/consent-config.js

```diff
@@ -150,7 +155,52 @@ export class ConsentConfig {
       }
     }
 
-    return config;
+    return this.mergeGeoOverride_(config).then(() => config);
+  }
+
+  /**
+   * Merge correct geoOverride object into toplevel config, then
+   * validate.
+   * @param {JsonObject} config
+   * @return {!Promise}
+   */
+  mergeGeoOverride_(config) {
+    if (config['geoOverride']) {
+      userAssert(
+        config['geoOverride']['geoGroupUnknown'],
+        'geoGroupUnknown must be specified in geoOverride'
+      );
+      Services.geoForDocOrNull(this.element_).then(geoService => {
+        userAssert(geoService, 'requires <amp-geo> to use geoOverride');
+        if (geoService.ISOCountry === 'unknown') {
+          config = /** @type {!JsonObject} */ (deepMerge(
+            config,
+            config['geoOverride']['geoGroupUnknown'],
+            1
+          ));
+        } else {
+          const geoGroups = Object.keys(config['geoOverride']);
+          for (let i = 0; i < geoGroups.length; i++) {
+            if (geoService.isInCountryGroup(geoGroups[i]) === GEO_IN_GROUP.IN) {
+              config = /** @type {!JsonObject} */ (deepMerge(
+                config,
+                config['geoOverride'][geoGroups[i]],
+                1
+              ));
+            }
+          }
+        }
+        delete config['geoOverride'];
+      });
+    }
+    if (config['consentRequired'] === 'remote') {
+      userAssert(
+        config['checkConsentHref'],
+        'checkConsentHref must be specified if consentRequired is remote'
+      );
+      config['consentRequired'] = true;
```

Comment:
why change it?

Reply:
@lannka  Should we just keep it `remote` and let it fail later down the line?

Reply:
the above userAssert will throw